cmd/gpu_fakedev/gpu_fakedev.go

```diff
@@ -302,6 +311,114 @@ func getOptions(name string) genOptions {
 	return opts
 }
 
+func makeXelinkSideCar(topology string, gpus int, tiles int, connections string) {
+	if topology != fullyConnected {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d)", gpus, tiles)
+	} else {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d, Topology: %s)", gpus, tiles, topology)
+	}
+
+	if topology == fullyConnected {
+		// create 2d slice with zeros
+		var cmap = make([][]int, 0)
```

Comment:
Because:
* This is for on/off info, it should be `bool` array, not `int` one
* Array size is known, it should be made with that size. You can then assign values directly by index instead of re-allocing array constantly with append()